src/eth.rs

```diff
@@ -103,14 +109,43 @@ impl EthereumClient {
         let block_number: [u8; 8] = block_number[24..].try_into().unwrap();
         let block_number = u64::from_be_bytes(block_number);
 
+        #[cfg(target_arch = "wasm32")]
+        {
+            let ms = now.elapsed().as_millis();
+            web_sys::console::log_1(
+                &format!("call to stateBlockNumber completed in {ms} ms")
+                    .into(),
+            );
+            #[allow(unused_variables)]
+            let now = Instant::now();
+        }
+
         let data = 0x382d83e3u32.to_be_bytes(); // keccak("stateBlockHash()")
         let block_hash: H256 =
             self.call(&data, tag).await.context("helios: state block hash")?;
 
+        #[cfg(target_arch = "wasm32")]
+        {
+            let ms = now.elapsed().as_millis();
+            web_sys::console::log_1(
+                &format!("call to stateBlockHash completed in {ms} ms").into(),
+            );
+            #[allow(unused_variables)]
+            let now = Instant::now();
+        }
```

Comment:
What I was referring in my comments is this part. In a simplified way, this could look like this:
![Simple](https://github.com/user-attachments/assets/4333c0b3-a289-40bf-bd38-50be09728916)

You have a `now` that has been defined at line 102 and another definition of `now` which seems to be local within the scope. Therefore, from my understanding, if this is supposed to be some checkpoint, then it should be outside of scope or declared as `mut` at the beginning. Otherwise, it can be deleted.

Please correct me if I am wrong somewhere.

Reply:
You are absolutely right, my bad.

I was merging standalone re-definitions into a previous block with logging, and totally missed that they're being moved into a scope, and even persuaded myself that clippy is wrong!